### PR TITLE
Update dependency management doc with offline testing

### PR DESCRIPTION
**Pull request by michael-berk**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/michael-berk/mlflow/pull/11322?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11322/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11322
```

</p>
</details>

### What changes are proposed in this pull request?
* Add CLI example for `update-pip-requirements`

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/model/dependencies.rst [hunk 1]
```diff
@@ -438,15 +445,15 @@ To do so, use the **pip-requirements-override** option to specify pip dependenci
         import mlflow
 
         mlflow.models.predict(
-            model_uri="runs:/<run_id>/model",
+            model_uri="runs:/<run_id>/<model_path>",
             input_data=<input_data>,
```
Comment (B-Step62) [inline docs/source/model/dependencies.rst:449]: Could you add similar 1~3 sentences description about the input data in this tab for Python?

File: docs/source/model/dependencies.rst [hunk 2]
```diff
@@ -438,15 +445,15 @@ To do so, use the **pip-requirements-override** option to specify pip dependenci
...
             pip_requirements_override=["opencv-python==4.8.0"],
         )
 
     .. code-tab:: bash
 
         mlflow models predict \
-            -m runs:/<run_id>/model \
+            -m runs:/<run_id>/<model_path> \
             -i <input_path> \
```
Comment (B-Step62) [inline docs/source/model/dependencies.rst:457]: Could you add a quick note that the file pointed by `input_path` has to be a JSON file that contains the prediction payload, or a CSV file for pandas Dataframe when the user specifies `--content-type csv`?

**Review by B-Step62 — APPROVED** [review]
LGTM with a few very minor comments!

File: docs/source/model/dependencies.rst (hunk 2 above)
Comment (michael-berk) [reply docs/source/model/dependencies.rst:457]: I updated the CLI docs. Does that work?

Comment (michael-berk) [reply docs/source/model/dependencies.rst:457]: Added a note here as well. Good call

File: docs/source/model/dependencies.rst [hunk 3]
```diff
@@ -438,21 +445,28 @@ To do so, use the **pip-requirements-override** option to specify pip dependenci
...
```
Comment (B-Step62) [inline docs/source/model/dependencies.rst:457]:
```suggestion
            -I <input_path> \
```
This was my mistake, but the docstring change below make me realize (the arg is case sensitive).

**Review by B-Step62 — APPROVED** [review]
LGTM!!

**Pull request merged**